source/daplink/interface/swd_host.c

```diff
@@ -1051,6 +1051,19 @@ uint8_t swd_set_target_state_sw(target_state_t state)
             osDelay(2);
             swd_set_target_reset(0);
             osDelay(2);
+        
+            // Power down
+            if (!swd_write_dp(DP_CTRL_STAT, 0x0)) {
+                return 0;
+            }
+            
+            // Wait until ACK is deasserted
+            do {
+                if (!swd_read_dp(DP_CTRL_STAT, &val)) {
+                    return 0;
+                }
+            } while ((val & (CSYSPWRUPACK | CDBGPWRUPACK)) == 1);
```

Comment:
ADIv6 says that we must not clear `CSYSPWRUPREQ` and `CDBGPWRUPREQ` at the same time.
ADIv5 says `CSYSPWRUPREQ` must not be set to 1 while `CDBGPWRUPREQ` is set to 0.

While DAPLink doesn't support any targets that use ADIv6 yet, it will happen soon with the Cortex-M55. So it would be best to go ahead and make this compatible by clearing `CSYSPOWERUPREQ` first, followed by `CDBGPWRUPREQ`.

Reply:
This change is not implemented yet as it could affect the power consumption, so it should be validated before adding it to this PR or change it in a different PR.

Reply:
When you split the clearing of these bits, please also add a comment stating that ADIv6 specifies that it must be done like that and to please not merge them back together. 😄

Reply:
I have updated it as per ADIv6. Could you review it?